ext/OptimizationEnzymeExt.jl

```diff
@@ -15,13 +15,47 @@ isdefined(Base, :get_extension) ? (using Enzyme) : (using ..Enzyme)
     end
 end
 
+function inner_grad(θ, bθ, f, p, args::Vararg{Any, N}) where N
+    Enzyme.autodiff_deferred(Enzyme.Reverse,
+        Const(firstapply),
+        Active,
+        Const(f),
+        Enzyme.Duplicated(θ, bθ),
+        Const(p),
+        Const.(args)...),
+    return nothing
+end
+
+function hv_f2_alloc(x, f, p, args...)
+    dx = Enzyme.make_zero(x)
+    Enzyme.autodiff_deferred(Enzyme.Reverse,
+        firstapply,
+        Active,
+        f,
+        Enzyme.Duplicated(x, dx),
+        Const(p),
+        Const.(args)...)
+    return dx
+end
+
+function inner_cons(x, p, num_cons, i)
+    res = zeros(eltype(x), num_cons)
+    f.cons(res, x, p)
+    return res[i]
+end
+
+function cons_f2(x, dx, fcons, p, num_cons, i)
+    Enzyme.autodiff_deferred(Enzyme.Reverse, inner_cons, Active, Enzyme.Duplicated(x, dx), Const(p), Const(num_cons), Const(i))
```

Comment:
```suggestion
    Enzyme.autodiff_deferred(Enzyme.Reverse, inner_cons, Active, fcons, Enzyme.Duplicated(x, dx), Const(p), Const(num_cons), Const(i))
```

Won't we need to zero and duplicate the function if it's a closure?

Reply:
I don't follow, can you add more detail what you mean by zeroing the function? It doesn't need to be duplicated it should be Const iiuc (done that in #60)